### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by gabenorthcoder**
Description:
GET 200:/api/articles/:article_id/comments, returns an array of comments for the given article_id, with the most recent comments first
Tests:
200: see that all properties are return and with the right data types
200: the most recent comments first

Error Handling:

404: Should return an error if the article_id is a valid datatype but does not exsis
400: Should return an error if the article_id is no valid datatype

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -118,10 +119,58 @@ describe("API's", () => {
...
+        .then(({ body }) => {
+          const { comments } = body;
+          expect(comments.length).toBe(11);
+          comments.forEach((comment) => {
+            expect(typeof comment.comment_id).toBe("number");
+            expect(typeof comment.votes).toBe("number");
+            expect(new Date(comment.created_at)).toBeInstanceOf(Date);
+            expect(typeof comment.author).toBe("string");
+            expect(typeof comment.body).toBe("string");
+            expect(typeof comment.article_id).toBe("number");
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:143]: Solid happy path test, great that you're checking the length to avoid false positives if we get an empty array 👍 
We can be a bit more specific about the article number here as we know all the comments will be for article 1

File: __tests__/app.test.js [hunk 2]
```diff
@@ -118,10 +119,58 @@ describe("API's", () => {
...
+        });
+    });
+    test("GET 200: Sould return an array of comments for an article_id with most recent comments first", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then(({ body }) => {
+          const { comments } = body;
+          expect(comments.length).toBe(11);
+          expect(comments).toBeSortedBy("created_at", { descending: true });
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:154]: Great use of jest-sorted 👍

File: __tests__/app.test.js [hunk 3]
```diff
@@ -118,10 +119,58 @@ describe("API's", () => {
...
+        });
+    });
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:156]: We need one more 200 test, to check that an existing article with 0 comments returns a 200 and an empty array 🔍

File: __tests__/app.test.js [hunk 4]
```diff
@@ -118,10 +119,58 @@ describe("API's", () => {
...
+    test("GET 400: Should return an error if the article_id is no valid datatype", () => {
+      return request(app)
+        .get("/api/articles/Invalid-Id/comments")
+        .expect(400)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe("Bad Request");
+        });
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:173]: Solid 400 and 404 tests, nice!

File: app/controller/comments-controller.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+const { checkArticleExists } = require("../model/check-articles");
+const { selectComments } = require("../model/select-comments");
+
+exports.getComments = (req, res, next) => {
+  const { article_id } = req.params;
+  Promise.all([selectComments(article_id), checkArticleExists(article_id)])
```
Comment (nc-katherine-hurst) [inline app/controller/comments-controller.js:6]: Great that you've implemented this to check the article exists, just need a test to check this functionality! 🧪

File: app/controller/comments-controller.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+    .then(([comments]) => {
+      res.status(200).send({ comments });
+    })
+    .catch((err) => {
+      console.log(err);
```
Comment (nc-katherine-hurst) [inline app/controller/comments-controller.js:12]: `console.log` begone 🗑️

File: app/model/select-comments.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+exports.selectComments = (article_id) => {
+  return db
+    .query(
+      `SELECT    comment_id,
+      votes,
+      created_at,
+      author,
+      body,
+      article_id 
+      FROM comments WHERE article_id = $1
```
Comment (nc-katherine-hurst) [inline app/model/select-comments.js:11]: Nice use of $ syntax to avoid SQL injection

File: endpoints.json [hunk 8]
```diff
@@ -26,5 +33,37 @@
...
+          "votes": 0,
+          "created_at": "2018-05-30T15:59:13.341Z",
+          "author": "JohnDoe",
+          "body": "Seafood substitutions are increasing",
+          "article_id": 1,
+          "topic": "cooking"
+        }
+      ]
+    }
   }
```
Comment (nc-katherine-hurst) [inline endpoints.json:68]: Brill!

**Review by nc-katherine-hurst — COMMENTED** [review]
Great stuff Gabe, just need one more test for some functionality you've already implemented 🔍
